src/WebJobs.Script.WebHost/Middleware/SystemTraceMiddleware.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.Extensions;
+using Microsoft.Azure.WebJobs.Script.WebHost.Security.Authentication;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    internal class SystemTraceMiddleware
+    {
+        private readonly ILogger _logger;
+        private readonly RequestDelegate _next;
+
+        public SystemTraceMiddleware(RequestDelegate next, ILogger<SystemTraceMiddleware> logger)
+        {
+            _logger = logger;
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            SetRequestId(context.Request);
+
+            var request = context.Request;
+            var details = new JObject
+            {
+                { "requestId", request.GetRequestId() },
+                { "method", request.Method.ToString() },
+                { "uri", request.Path.ToString() }
+            };
+            var logData = new Dictionary<string, object>
+            {
+                [ScriptConstants.LogPropertyActivityIdKey] = request.GetRequestId()
+            };
+            _logger.Log(LogLevel.Information, 0, logData, null, (s, e) => $"Executing HTTP request: {details}");
+
+            await _next.Invoke(context);
+
+            details["identities"] = GetIdentities(context);
+            details["status"] = context.Response.StatusCode.ToString();
```

Comment:
Could we log the duration as well? I believe the default logger does that -- but could be mistaken.

Reply:
Yes I can do this. What do you mean the default logger does this? I'm planning on using my own simple timer

Reply:
The default logs that you get with ASP.NET (which we're hiding). I'll dig up a link to where they do this...

Reply:
I just wanted to make sure we weren't missing anything from the default logging experience and that we could reuse stuff if it already existed-- they do log duration but all of that looks internal. So using our own stopwatch works:

https://github.com/aspnet/Hosting/blob/master/src/Microsoft.AspNetCore.Hosting/Internal/HostingApplicationDiagnostics.cs#L109
https://github.com/aspnet/Hosting/blob/master/src/Microsoft.AspNetCore.Hosting/Internal/HostingRequestFinishedLog.cs#L53

Reply:
@brettsam just want clarify... when you say _which we're hiding_, is that something we're doing explicitly? The reason why we didn't have these HTTP traces ported over right away was because we wanted to inspect what we got out of the box to evaluate the need for this.

Reply:
We filter all logs that aren't ours (by looking at the category) -- https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Extensions/ScriptLoggingBuilderExtensions.cs#L36

Unfortunately, the default logs we'd get would include the query strings. We could hide that log specifically, but @mathewc and I were concerned that we'd be constantly chasing down logs that were sending too much info by default. By locking it down and only sending ours, we can now customize this and worry less about stuff sneaking by.